### DISCUSSION TITLE
Different queries for scaling up and scaling down

### DISCUSSION THREAD

**Discussion by Mikola3** [kedacore/keda#939] [General] [answered]
Dear KEDA team,

Could you please tell me if it is possible to have different queries for scaling up and scaling down within ScaleObject config?
For instance, I would like to use the following queries:

**Scale up**:

```
      threshold: '3'
      query: sume(rate(flask_request_count{endpoint="/",http_status="200",method="GET"}[2m]))
```

**Scale down**:
```
      threshold: '3'
      query: sume(rate(flask_request_count{endpoint="/",http_status="200",method="GET"}[5m]))
```

This will help avoid scaling down in case of shape spikes.

Thank you in advance for your feedback!

**Comment by zroubalik** [comment, marked as answer]
Hi @Mikola3 

unfortunately that's not possible and I am not sure if that's something that we could try to support in the future. The scaling is internally dependent on HPA, so that would be something that would be must be supported there first.

For our upcoming KEDA v2, we are adding capabilites to tweak the scaling behavior (in k8s > v1.18) as it is something that was brought to HPA:https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-configurable-scaling-behavior you can take a look on that and maybe it helps with your issue.

**Reply by Mikola3** [reply]
@zroubalik, thank you for your answer.

Let me clarify one additional thing about scaling down.

I wonder if there is any graceful period of time for scaling down pods. It will allow existing connections not to be terminated immediately. As a result, the user will not see a sudden error in this case.

**Reply by tbickford** [reply]
@Mikola3 If your deployment has the `terminationGracePeriodSeconds` set (and/or any preStop hooks) to handle graceful connection termination, that would still be respected with or without KEDA.  In other words it sounds like the problem you are facing is more so related to how the deployment is setup to handle termination.

**Reply by Mikola3** [reply]
@tbickford, @zroubalik thank you for your assistance. Your answers are really helpful.
